src/components/labelPrinting/LabelPrintingForm.vue

```diff
@@ -112,114 +95,74 @@
-        let barcodes = this.form.barcode.split(/\r?\n|\r|\n/g)
-        switch (true) {
-          case this.suffix() && applyLabels:
-            //Add both suffix and label number to the barcodes to be printed
-            listSuffixedBarcodes = this.appendSuffixWithLabels(
-              barcodes,
-              listSuffixedBarcodes,
-              noOfLabels,
-            )
-            break
-          case this.suffix() && !applyLabels:
-            //Add only suffix to the barcodes to be printed
-            listSuffixedBarcodes = this.appendSuffix(barcodes, listSuffixedBarcodes)
-            break
-          case applyLabels && !this.suffix():
-            //Add only label number to the barcodes to be printed
-            listSuffixedBarcodes = this.appendLabels(barcodes, listSuffixedBarcodes, noOfLabels)
-            break
-          case !applyLabels && !this.suffix():
-            //Just the plain barcodes list to be printed
-            for (let barcode of barcodes) {
-              listSuffixedBarcodes.push(barcode)
-            }
-            break
-        }
-      }
-      return listSuffixedBarcodes
     },
-    appendSuffix(barcodes, listSuffixedBarcodes) {
-      for (let barcode of barcodes) {
-        listSuffixedBarcodes.push(barcode.concat('-', this.suffix()))
-      }
-      return listSuffixedBarcodes
+    suffixOptions() {
+      return createSuffixDropdownOptions(SuffixList)
     },
-    appendLabels(barcodes, listSuffixedBarcodes, noOfLabels) {
-      for (let barcode of barcodes) {
-        for (let i = 0; i < noOfLabels; i++) {
-          listSuffixedBarcodes.push(barcode.concat('-', i + 1))
-        }
-      }
-      return listSuffixedBarcodes
+    suffixItems() {
+      return createSuffixItems(SuffixList)
     },
-    appendSuffixWithLabels(barcodes, listSuffixedBarcodes, noOfLabels) {
-      for (let barcode of barcodes) {
-        for (let i = 0; i < noOfLabels; i++) {
-          listSuffixedBarcodes.push(barcode.concat('-', this.suffix(), '-', i + 1))
-        }
-      }
-      return listSuffixedBarcodes
-    },
-    suffix() {
-      return this.form.selectedSuffix === 'No suffix' ? '' : this.form.selectedSuffix.slice(0, 4)
-    },
-    printerName() {
-      return this.form.selectedPrinterName
+    labels() {
+      const date = getCurrentDate()
+      const suffixItem = this.suffixItems[this.form.suffix]
+
+      // it is possible for there to be no barcodes so we need to add a guard
+      const splitSourceBarcodeList = this.form.sourceBarcodeList?.split(/\r?\n|\r|\n/g) || []
+
+      return createLabelsFromBarcodes({
+        sourceBarcodeList: splitSourceBarcodeList,
+        date,
+        suffixItem,
+        numberOfLabels: this.form.numberOfLabels,
+      })
     },
-    onSubmit(event) {
+  },
+  methods: {
+    /*
+      Creates the print job and shows a success or failure alert
+      @param {event}
+    */
+    async printLabels(event) {
```

Comment:
How come `event.preventDefault()` is needed?

Reply:
Because it is a form so the submit button on a form the default would be to refresh the page or move to another one.